gh-129005: Avoid copy in _pyio.FileIO.readinto (#129006)
python/cpython: Misc/NEWS.d/next/Library/2025-01-18-15-29-08.gh-issue-129005.mM5dmV.rst

> Change ``_pyio.FileIO.readinto`` to use :func:`os.readv` avoiding a copy that was present when using :func:`os.read` on platforms with :func:`os.readv`

Reviewer: ```suggestion
Optimize :meth:`_pyio.FileIO.readinto` by using :func:`os.readv` on platforms that support it which avoids an additional copy.
```

Author: Looks like `:meth:` doesn't work because `_pyio` isn't in the public docs so Sphinix emits a warning. Adjusting that piece.

Reviewer: Yep, seems to be the case. Thanks!
